src/Pages/EventPage/EventPage.jsx

```diff
@@ -0,0 +1,16 @@
+import { useParams } from "react-router-dom";
+import { Event } from "../../Components";
+import EventData from "../../Data/EventData.json";
+
+const EventPage = () => {
+  const eventID = useParams();
+  const ind = parseInt(eventID.id, 10);
+  return (
+    <>
+      {ind > 0 && <Event data={EventData[ind - 1]} />}
+      {ind <= 0 && <div>Error Page</div>}
+    </>
+  );
+};
```

Comment:
Handle the logic properly if the index doesn't exist.

There are many cases, like if index < 0 or index > length of the array, also there may be the case that id are not serially, so all the cases need to be handled properly. 

Better to keep the events data in `public/db` and then fetch the event with `id = eventID.id` and if not found then redirect it to `/` using the `useNavigate` hook of react-router.